posttroll/subscriber.py

```diff
@@ -455,28 +340,29 @@ def create_subscriber_from_dict_config(settings):
     :class:`~posttroll.subscriber.Subscriber` and :class:`~posttroll.subscriber.NSSubscriber`.
 
     """
-    if settings.get('addresses') and settings.get('nameserver') is False:
+    if settings.get("addresses") and settings.get("nameserver") is False:
         return _get_subscriber_instance(settings)
     return _get_nssubscriber_instance(settings).start()
 
 
 def _get_subscriber_instance(settings):
-    addresses = settings['addresses']
-    topics = settings.get('topics', '')
-    message_filter = settings.get('message_filter', None)
-    translate = settings.get('translate', False)
+    _ = settings.pop("nameserver", None)
+    _ = settings.pop("port", None)
+    _ = settings.pop("services", None)
+    _ = settings.pop("addr_listener", None),
+    _ = settings.pop("timeout", None)
 
-    return Subscriber(addresses, topics=topics, message_filter=message_filter, translate=translate)
+    return Subscriber(**settings)
```

Comment:
I think the original is cleaner as well as explicit. If there are extra items in `settings` the `Subscriber` creation will fail.

Reply:
Yes, that's the point. Until now we could pass whatever and not even know it was wrong...